prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/RoutingTests.kt

```diff
@@ -102,17 +125,146 @@ class RoutingTests {
+        val settings = FileSettings().loadOrganizations(twoOrganization)
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        // act
+        val result = engine.replaceShorthand("%test-dash")
+
+        // assert
+        assertThat(result).isEqualTo("Bundle.test.dash")
+    }
+
+    @Test
+    fun `test replaceShorthand with underscored string`() {
+        val settings = FileSettings().loadOrganizations(twoOrganization)
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        // act
+        val result = engine.replaceShorthand("%test_underscore")
+
+        // assert
+        assertThat(result).isEqualTo("Bundle.test.underscore")
+    }
+
+    @Test
+    fun `test replaceShorthand with apostrophe string`() {
+        val settings = FileSettings().loadOrganizations(twoOrganization)
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        // act
+        val result = engine.replaceShorthand("%test'apostrophe")
+
+        // assert
+        assertThat(result).isEqualTo("Bundle.test.apostrophe")
+    }
+
+    @Test
+    fun `test replaceShorthand with multi parts string`() {
+        val settings = FileSettings().loadOrganizations(twoOrganization)
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        // act
+        val result = engine.replaceShorthand("%patient.name.first")
+
+        // assert
+        assertThat(result).isEqualTo("Bundle.entry.resource.ofType(Patient).name.first")
+    }
+
+    @Test
+    fun `test replaceShorthand with multi part while in middle`() {
+        val settings = FileSettings().loadOrganizations(twoOrganization)
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        // act
+        val result = engine.replaceShorthand("myVariable.%patient.name.first")
+
+        // assert
+        assertThat(result).isEqualTo("myVariable.Bundle.entry.resource.ofType(Patient).name.first")
+    }
+
+    @Test
+    fun `test replaceShorthand with similar strings, order 1`() {
+        val settings = FileSettings().loadOrganizations(twoOrganization)
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        // act
+        val result = engine.replaceShorthand("%patient or %patientState")
+
+        // assert
+        assertThat(result).isEqualTo(
+            "Bundle.entry.resource.ofType(Patient) or Bundle.entry.resource.ofType(Patient).address.state"
+        )
+    }
+
+    @Test
+    fun `test replaceShorthand with similar strings, order 2`() {
+        val settings = FileSettings().loadOrganizations(twoOrganization)
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        // act
+        val result = engine.replaceShorthand("%patientState or %patient")
```

Comment:
Add similar tests to cover a sample of
%'var'
%\`var`
I did this and it failed for me:
`engine.replaceShorthand("%'patientState' or %`patient`")`
Error: `Could not locate a unique entry for '%'patientState'' in the shorthand values lookup table.`